server/src/main/java/org/elasticsearch/action/FailedNodeException.java

```diff
@@ -43,4 +44,25 @@ protected void writeTo(StreamOutput out, Writer<Throwable> nestedExceptionsWrite
     protected void metadataToXContent(XContentBuilder builder, Params params) throws IOException {
         builder.field("node_id", nodeId);
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        FailedNodeException that = (FailedNodeException) o;
+        return Objects.equals(nodeId, that.nodeId)
+            && getMessage().equals(that.getMessage())
+            && getStackTrace().length == that.getStackTrace().length; // Could check elements individually, though there can be slight
+                                                                      // differences when serialized.
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(
+            nodeId,
+            getMessage(),
+            getClass().getName(),
+            getStackTrace().length // Could check elements individually, though there can be slight differences when serialized.
+        );
```

Comment:
I am wondering if these changes should go to their own PR for visibility, as they have little to do with evicting all items from the blob cache. (I do remember why you needed it in the first place, and a separate PR would highlight the reson why you need it and solicit targeted feedback for that problem.)